pkg/engine/mutate/generatePatches.go

```diff
@@ -0,0 +1,27 @@
+package mutate
+
+import (
+	"fmt"
+
+	"github.com/mattbaird/jsonpatch"
+)
+
+func generatePatches(src, dst []byte) ([][]byte, error) {
+	var patchesBytes [][]byte
+	pp, err := jsonpatch.CreatePatch(src, dst)
+	for _, p := range pp {
+		// TODO: handle remove nil value, i.e.,
```

Comment:
what does this mean?

Reply:
For example, if security context is not defined in resource, the tag `securityContext` is preserved in resource manifest, carried along with admission request (`src` in this case), while this tag is dropped in strategic-merge-patched document (`dst` in this case). When comparing two documents, it will generate a remove JSON patch - `{remove /spec/securityContext <nil>}`. It does not impact the final result though, would be good to find a way to ignore these patches.